Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m

```diff
@@ -222,6 +224,36 @@ - (void)removeWebView:(WKWebView *)webView
     [self.tabViewController.tableView reloadData];
 }
 
+- (NSString *)addProtocolIfNecessary:(NSString *)address
+{
+    if ([address rangeOfString:@"://"].length > 0)
+        return address;
+
+    if ([address hasPrefix:@"data:"])
+        return address;
+
+    if ([address hasPrefix:@"about:"])
+        return address;
+
+    return [@"http://" stringByAppendingString:address];
+}
+
+- (NSURL *)targetURLorDefaultURL
+{
+    NSArray *args = [[NSProcessInfo processInfo] arguments];
+    const NSUInteger targetURLIndex = [args indexOfObject:kURLArgumentString];
+
+    // FIXME: Add support for passing file URLs on the command line.
+    if (targetURLIndex != NSNotFound && targetURLIndex + 1 < [args count]) {
+        NSString *targetURL = [self addProtocolIfNecessary:[args objectAtIndex:targetURLIndex + 1]];
+        NSURL *url = [NSURL URLWithString:targetURL];
```

Comment:
Might be nicer to say:

```
if (NSURL *url = [NSURL URLWithString: targetURL])
    return url;
```

Reply:
I tried that! I know you can do it in C(++). This is what Xcode yells at me with: `Unexpected interface name 'NSURL': expected expression`

I also tried wrapping the assignment in parens and no go.